model/task/task.go

```diff
@@ -1988,6 +1988,45 @@ func GetTimeSpent(tasks []Task) (time.Duration, time.Duration) {
 	return timeTaken, latestFinishTime.Sub(earliestStartTime)
 }
 
+// GetTaskResultsByVersion gets all tasks for a specific version
+// Query results can be filtered by task name, variant name and status in addition to being paginated and limited
+func GetTaskResultsByVersion(versionID, taskOrVariantName, sortBy string, statuses []string, sortDir, page, limit int) ([]Task, error) {
+	tasks := []Task{}
+	match := bson.M{
+		VersionKey: versionID,
+	}
+	pipeline := []bson.M{
+		{"$match": &match},
+	}
+	if len(statuses) > 0 {
+		statusFilters := []bson.M{}
+		for _, status := range statuses {
+			statusFilters = append(statusFilters, bson.M{StatusKey: status})
+		}
+		match["$or"] = statusFilters
+	}
+	if len(taskOrVariantName) > 0 {
+		match["$or"] = []bson.M{
```

Comment:
Let's avoid overloading this parameter both for clarity and for performance. I assume this is for the search box that searches tasks or variants, but that design is currently being debated